data-runtime/src/main/java/io/micronaut/data/runtime/mapper/sql/SqlResultEntityTypeMapper.java

```diff
@@ -590,7 +590,8 @@ private <K> K readEntity(RS rs, MappingContext<K> ctx, @Nullable Object parent,
                         entity = setProperty(property, entity, value);
                     } else {
                         final boolean isInverse = parent != null && entityAssociation.getKind().isSingleEnded() && isAssociation && ctx.association.getOwner() == entityAssociation.getAssociatedEntity();
-                        if (isInverse) {
+                        // Before setting property value, check if mappedBy is not different from the property name
+                        if (isInverse && mappedByMatchesOrEmpty(ctx.association, property)) {
                             entity = setProperty(property, entity, parent);
```

Comment:
Without this check, parent entity would be set to all one-to-many collections, disregarding mappedBy attribute on the relation. So, this adds check if mappedBy is set and is matching with the field then can set it.